lti_consumer/lti_consumer.py

```diff
@@ -723,6 +718,32 @@ def is_past_due(self):
             close_date = due_date
         return close_date is not None and timezone.now() > close_date
 
+    def _get_lti1p1_consumer(self):
+        """
+        Returns a preconfigured LTI 1.1 consumer.
+
+        If the block is configured to use LTI 1.1, set up a
+        base LTI 1.1 consumer class.
+        This class does NOT store state between calls.
+        """
+        key, secret = self.lti_provider_key_secret
+        return LtiConsumer1p1(self.launch_url, key, secret)
+
+    def extract_real_user_data(self):
+        """
+        Set actual user data on xblock from the runtime
+        """
+        if callable(self.runtime.get_real_user):
+            real_user_object = self.runtime.get_real_user(self.runtime.anonymous_student_id)
+            self.user_email = getattr(real_user_object, "email", "")
+            self.user_username = getattr(real_user_object, "username", "")
+            user_preferences = getattr(real_user_object, "preferences", None)
+
+            if user_preferences is not None:
+                language_preference = user_preferences.filter(key='pref-lang')
+                if len(language_preference) == 1:
+                    self.user_language = language_preference[0].value
```

Comment:
Instead of storing these in self, just return them from `extract_real_user_data`?

```suggestion
    def extract_real_user_data(self):
        """
        Set actual user data on xblock from the runtime
        """
        user_email = None
        user_username = None
        user_language = None
        if callable(self.runtime.get_real_user):
            real_user_object = self.runtime.get_real_user(self.runtime.anonymous_student_id)
            user_email = getattr(real_user_object, "email", "")
            user_username = getattr(real_user_object, "username", "")
            user_preferences = getattr(real_user_object, "preferences", None)

            if user_preferences is not None:
                language_preference = user_preferences.filter(key='pref-lang')
                if len(language_preference) == 1:
                    user_language = language_preference[0].value
        return user_email, user_username, user_lanauge
```

Reply:
I've implemented something similar to this in the most recent commit